libs/evcs-service/src/main/java/uk/gov/di/ipv/core/library/client/EvcsClient.java

```diff
@@ -138,31 +142,27 @@ public void updateEvcsUserVCs(String userId, List<EvcsUpdateUserVCsDto> evcsUser
 
     private URI getUri(String userId, List<EvcsVCState> vcStatesToQueryFor)
             throws URISyntaxException {
+        var uriBuilder =
+                new URIBuilder(configService.getSsmParameter(EVCS_APPLICATION_URL))
+                        .setPathSegments("vcs", userId);
         if (vcStatesToQueryFor != null) {
-            return new URIBuilder(
-                            configService
-                                    .getSsmParameter(EVCS_APPLICATION_URL)
-                                    .concat("/")
-                                    .concat(userId))
-                    .addParameter(
-                            VC_STATE_PARAM,
-                            vcStatesToQueryFor.stream()
-                                    .map(EvcsVCState::name)
-                                    .collect(Collectors.joining(",")))
-                    .build();
-        } else {
-            return new URIBuilder(
-                            configService
-                                    .getSsmParameter(EVCS_APPLICATION_URL)
-                                    .concat("/")
-                                    .concat(userId))
-                    .build();
+            uriBuilder.addParameter(
+                    VC_STATE_PARAM,
+                    vcStatesToQueryFor.stream()
+                            .map(EvcsVCState::name)
+                            .collect(Collectors.joining(",")));
         }
+        return uriBuilder.build();
     }
 
     private void checkResponseStatusCode(HttpResponse<String> evcsHttpResponse)
             throws EvcsServiceException {
         if (200 > evcsHttpResponse.statusCode() || evcsHttpResponse.statusCode() > 299) {
+            LOGGER.info(
+                    LogHelper.buildLogMessage(
+                            ErrorResponse.RECEIVED_NON_200_RESPONSE_STATUS_CODE.getMessage()
+                                    + String.format(
+                                            " (%s) received", evcsHttpResponse.statusCode())));
```

Comment:
An even better way of doing this would be to make it a log property:
```
LOGGER.info(
        LogHelper.buildLogMessage(
                        ErrorResponse.RECEIVED_NON_200_RESPONSE_STATUS_CODE
                                .getMessage())
                .with(LOG_STATUS_CODE.getFieldName(), evcsHttpResponse.statusCode()));
```
This would make it queryable, so we could (e.g.) easily get metrics on which status codes were being received.

Would you be able to add this to the corresponding place in the TICF service as well, for completeness?

Reply:
done ...updated ticf service also for log statement.

Reply:
Thanks, that's really helpful!